aggregators/rules/cr.js

```diff
@@ -1,16 +1,96 @@
+var request = require("request");
+var console = require("console");
+
+const CR_ADDRESS = process.env.CR_ADDRESS; // http://media.wizards.com/2016/docs/MagicCompRules_20160930.txt or newer
+
 class CR{
     constructor(){
-        this.Location = "http://blogs.magicjudges.org/rules/cr/";
+		this.location = "http://blogs.magicjudges.org/rules/cr/";
+		request({url: CR_ADDRESS, encoding: "utf16le"}, (error, response, body) => {
+			 if (!error && response.statusCode == 200) {
+				 let {cr, glossary} = parseCR(body);
+				 this.cr = cr;
+				 this.glossary = glossary;
+				 console.log("CR Ready")
+			 } else {
+				 console.error(error);
+			 }
+		});
     }
     find (parameter,callback){
-        //todo
+        callback(this.cr.get(parameter.trim().replace(/\.$/, "").toLowerCase()));
     }
     getContent(parameter,callback) {
         if (parameter) {
-            this.find(parameter, callback());
+            this.find(parameter, callback);
         } else {
-            callback(this.Location);
+            callback(this.location);
         }
     }
+	getGlossaryEntry(parameter, callback) {
+		callback(this.glossary.get(parameter.trim().toLowerCase()));
+	}
 }
+
+
+function parseCR(crText) {
+	crText = crText.replace(/\r/g, "");
+
+	let rulesText = crText.substring(crText.search("\n\n100. General\n") + 2, crText.length);
```

Comment:
`rulesText` gets rebinded in line 42. So when you change it you have to use a additional constant,